### PR TITLE
Even more ES6 ports

### PR DESCRIPTION
**Pull request by jwietelmann**

### CODE REVIEW

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const value = e.target[this.isCheckbox() ? 'checked' : 'value']
+
+    this.props.model.set(this.props.field, value)
+
+    if(this.props.onChange) {
+      this.props.onChange(e)
+    }
+  },
+
+  onBlur(e) {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: "e" is defined but never used

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const validator = this.props.model[`validate${_.str.classify(this.props.field)}`]
+
+    if(validator) {
+      const error = validator.bind(this.props.model)()
+
+      if(error) {
+        const errors = {}
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: Expected blank line after variable declarations.

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        errors[this.props.field] = [error]
+        const response = {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: Expected blank line after variable declarations.

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+          responseJSON: {errors}
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: Missing trailing comma.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6 [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        {this.props.children}
+      </Form>
+    )
+  }
+
+  formAction() {
+    return this.props.action || this.props.model.url()
+  }
+
+  formMethod() {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6]: Function 'formMethod' has a complexity of 4.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6 [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    }
+  }
+
+  render() {
+    const formParams = {
+      defaultValue: this.defaultValue(),
+      name: this.name(),
+    }
+
+    const type = ['textarea', 'select'].find((type) => type === this.props.type)
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6]: "type" is already declared in the upper scope.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6 [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+    if(this.isCheckbox()) {
+      formParams['defaultChecked'] = this.value()
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6]: ["defaultChecked"] is better written in dot notation.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 8]
```diff
@@ -0,0 +1,86 @@
+Rev.registerComponent('ModelProgressIndicator', class ModelProgressIndicator extends React.Component {
+
+  static get Status() {
+    return this._Status = this._Status || {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Return statement should not contain assignment.

Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Unexpected `this`.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      status: this.constructor.Status.STARTED,
+    }
+
+    this.onRequest = this.onRequest.bind(this)
+    this.onSync = this.onSync.bind(this)
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Unexpected sync method: 'onSync'.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  render() {
+    return (
+      <span className={this.progressClasses()}>
+        {this.childForCurrentStatus()}
+      </span>
+    )
+  }
+
+  childForCurrentStatus() {
+    const statusNames = ['started', 'requested', 'synced', 'errored']
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected blank line after variable declarations.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    for(let i = 0; i < statusNames.length; i++) {
+      const currentStatus = this.childForStatus(statusNames[i])
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected blank line after variable declarations.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+      if(currentStatus) {
+        return currentStatus
+      }
+    }
+  }
+
+  childForStatus(statusName) {
+    if(this.props[statusName] && this.state.status == this.constructor.Status[statusName.toUpperCase()]) {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected '===' and instead saw '=='.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+      return this.props[statusName]
+    }
+  }
+
+  componentDidMount() {
+    this.props.model.on('request', this.onRequest)
+    this.props.model.on('sync', this.onSync)
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Unexpected sync method: 'onSync'.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    this.props.model.on('error', this.onError)
+  }
+
+  componentWillUnmount() {
+    this.props.model.off('request', this.onRequest)
+    this.props.model.off('sync', this.onSync)
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Unexpected sync method: 'onSync'.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+  onError() {
+    this.setState({
+      status: this.constructor.Status.ERRORED,
+    })
+  }
+
+  progressClasses() {
+    return this.suitSet('ModelProgressIndicator', {
+      started: this.state.status == this.constructor.Status.STARTED,
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected '===' and instead saw '=='.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 16]
```diff
@@ -0,0 +1,86 @@
...
+      requested: this.state.status == this.constructor.Status.REQUESTED,
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected '===' and instead saw '=='.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+      synced: this.state.status == this.constructor.Status.SYNCED,
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected '===' and instead saw '=='.

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+      errored: this.state.status == this.constructor.Status.ERRORED,
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected '===' and instead saw '=='.

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 19]
```diff
@@ -0,0 +1,55 @@
+Rev.registerMixin('ModelInputMixin', {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: Expected changes or additions to a test file called [ModelInputMixin_spec.js](https://github.com/revelrylabs/revelry_core/blob/eb8fc080150ef92737e8c27bdbffcf4bba876528/spec/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin_spec.js)

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6 [hunk 20]
```diff
@@ -0,0 +1,76 @@
+// A form that works with Backbone models to do AJAXy submission & backbone
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6]: Expected changes or additions to a test file called [ModelForm_spec.js](https://github.com/revelrylabs/revelry_core/blob/eb8fc080150ef92737e8c27bdbffcf4bba876528/spec/javascripts/revelry/ui/forms/ModelForm/ModelForm_spec.js)

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6 [hunk 21]
```diff
@@ -0,0 +1,74 @@
+Rev.registerComponent('ModelInput', class ModelInput extends React.Component {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6]: Expected changes or additions to a test file called [ModelInput_spec.js](https://github.com/revelrylabs/revelry_core/blob/eb8fc080150ef92737e8c27bdbffcf4bba876528/spec/javascripts/revelry/ui/forms/ModelForm/ModelInput_spec.js)

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 22]
```diff
@@ -0,0 +1,86 @@
+Rev.registerComponent('ModelProgressIndicator', class ModelProgressIndicator extends React.Component {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected changes or additions to a test file called [ModelProgressIndicator_spec.js](https://github.com/revelrylabs/revelry_core/blob/eb8fc080150ef92737e8c27bdbffcf4bba876528/spec/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator_spec.js)

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.herokuapp.com/relint/revelrylabs/revelry_core/87).

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6 (hunk 5 above)
Comment (jwietelmann) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6]: Seriously???

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+    const value = e.target[this.isCheckbox() ? 'checked' : 'value']
+
+    this.props.model.set(this.props.field, value)
+
+    if(this.props.onChange) {
+      this.props.onChange(e)
+    }
+  },
+
+  onBlur(_e) {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: Identifier name '_e' is too short. (< 3)

File: app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6 [hunk 24]
```diff
@@ -0,0 +1,58 @@
+Rev.registerMixin('ModelInputMixin', {
```
Comment (lintomat) [inline app/assets/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin.es6]: Expected changes or additions to a test file called [ModelInputMixin_spec.js](https://github.com/revelrylabs/revelry_core/blob/2470832b297fa22124aeec717ac0920a032a204f/spec/javascripts/revelry/mixins/ui/forms/ModelForm/ModelInputMixin_spec.js)

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6 (hunk 20 above)
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelForm.es6]: Expected changes or additions to a test file called [ModelForm_spec.js](https://github.com/revelrylabs/revelry_core/blob/2470832b297fa22124aeec717ac0920a032a204f/spec/javascripts/revelry/ui/forms/ModelForm/ModelForm_spec.js)

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6 (hunk 21 above)
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelInput.es6]: Expected changes or additions to a test file called [ModelInput_spec.js](https://github.com/revelrylabs/revelry_core/blob/2470832b297fa22124aeec717ac0920a032a204f/spec/javascripts/revelry/ui/forms/ModelForm/ModelInput_spec.js)

File: app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6 [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
```
Comment (lintomat) [inline app/assets/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator.es6]: Expected changes or additions to a test file called [ModelProgressIndicator_spec.js](https://github.com/revelrylabs/revelry_core/blob/2470832b297fa22124aeec717ac0920a032a204f/spec/javascripts/revelry/ui/forms/ModelForm/ModelProgressIndicator_spec.js)

**Pull request merged**